### PR TITLE
Sql lesson2

### PR DESCRIPTION
**Pull request by ohasselblad**
@auremoser @andrewxhill @michellechandra 

Would you mind looking over? 

Pretty close to final form. 

p.s., if you want to run it locally check out the readme for the repo.

### CODE REVIEW

File: courses/04-sql-postgis/lesson-2.md [hunk 1]
```diff
@@ -2,12 +2,307 @@
...
 
-## Editing Tables, more advanced PostGIS -- Coming Soon
+## PostGIS in CartoDB
+
+CartoDB is built on top of [PostgreSQL](http://www.postgresql.org/) using the [PostGIS](http://www.postgresql.org/) extension. This means that you have all the power of relational databases combined with hundreds of geospatial functions. 
+
+In this lesson, we will introduce several commonly used functions in PostGIS with the goal of extending your geospatial analysis of data within CartoDB and show you some of the analysis you can do with your geospatial data.
+
+**Our goal with this lesson:**
+Geospatially visit blues musician birthplaces along Highway 61 by using important functions in PostGIS and SQL.
```
Comment (michellechandra) [inline courses/04-sql-postgis/lesson-2.md]: Include more info about Highway 61 and it’s importance (briefly) for people who don’t know what Highway 61 is (and the relationship to blues music.)

File: courses/04-sql-postgis/lesson-2.md [hunk 2]
```diff
@@ -2,12 +2,307 @@
...
+
+_If you haven't gotten your feet wet with SQL in CartoDB, check out [Lesson 1](http://academy.cartodb.com/courses/04-sql-postgis/lesson-1.html) first._
```
Comment (michellechandra) [inline courses/04-sql-postgis/lesson-2.md]: Maybe emphasize here that users will use the CartoDB editor entirely for this lesson? So should have an account and be familiar with the editor to import data and run SQL queries?

File: courses/04-sql-postgis/lesson-2.md [hunk 3]
```diff
@@ -2,12 +2,307 @@
...
+The second data set contains the coordinates of the birthplaces of a selection of Mississippi-born blues musicians as [listed on Wikipedia](http://en.wikipedia.org/wiki/List_of_Delta_blues_musicians).
+
+{% highlight text %}
+http://academy.cartodb.com/d/mississippi_blues_musicians.geojson
+{% endhighlight %}
+
+
+### Show Buffers
+
+As the name suggests, `ST_Buffer` takes a geometry as an argument and returns that geometry but with a buffer around it. It always returns a polygon. In its simplest usage, you can create a polygon centered on a point. The same applies to lines and polygons, but you always get out a polygon.
```
Comment (michellechandra) [inline courses/04-sql-postgis/lesson-2.md]: Define what buffers are in the GIS/geography space and link to more info such as this Wiki article: http://en.wikipedia.org/wiki/Buffer_%28GIS%29

File: courses/04-sql-postgis/lesson-2.md [hunk 4]
```diff
@@ -2,12 +2,307 @@
...
+
+Like other PostGIS functions, it preserves the projection units. That is, if you put in a point in WGS 84, you'll get out a polygon in WGS 84.
```
Comment (michellechandra) [inline courses/04-sql-postgis/lesson-2.md]: Clarify what WGS 84 is for people who aren't familiar.

File: courses/04-sql-postgis/lesson-2.md [hunk 5]
```diff
@@ -2,12 +2,307 @@
...
+{% highlight c++ %}
+geometry ST_Buffer(geometry g1, float radius_of_buffer)
+geography ST_Buffer(geography g1, float radius_of_buffer_in_meters)
+{% endhighlight %}
+
+The function 'ST_Buffer' takes a geometry such as a point as the first argument, and the radius for the buffer as the second argument. For more inforation, see the PostGIS [documentation page](http://www.postgis.org/docs/ST_Buffer.html).
+
+There is a hidden column in CartoDB called `the_geom_webmercator,` which is responsible for making the data appear on your map in the correct location. Its units are meters and we will be using it exclusively. It is in the projection [WebMercator](http://en.wikipedia.org/wiki/Web_Mercator), or [ESPG 3857](http://spatialreference.org/ref/sr-org/7483/). `the_geom_webmercator` is updated in the background every time you update `the_geom`. It is _not_ updated, though, when you only select `the_geom` in a `SELECT` statement.
+
+To visualize a 25 mile corridor around U.S. Route 61, put the following command into the cartoDB [SQL editor](http://docs.cartodb.com/cartodb-editor.html#custom-sql).
```
Comment (michellechandra) [inline courses/04-sql-postgis/lesson-2.md]: Wondering if should clarify here that after uploading the tables, users should navigate to the map view of the highway 61 table, open the sql editor in the cartoDB sidebar and run the following commands?

**Comment by andrewxhill** [comment]
hey @michellechandra for some of those, don't be bashful about going ahead and making the changes that you think will improve it. unless andy told you not to, then i take it back

File: courses/04-sql-postgis/lesson-2.md [hunk 6]
```diff
@@ -2,12 +2,307 @@
...
+{% endhighlight %}
+
+This command grabs all locations, names, cities, and cartodb_ids from rows in the `mississippi_blues_musicians` table that are within a 25 mile distance of the highway. Recall from [Lesson 1](http://academy.cartodb.com/courses/04-sql-postgis/lesson-1.html) that the `AS` keyword lets you define table names and column names with a more concise or less confusing alias.
+
+The embedded map below has the bottom layer as the buffered highway, and the top layer as the SQL statement applied to the `mississippi_blues_musicians` table to only show the musicians within our buffer.
+
+<iframe width='100%' height='520' frameborder='0' src='http://documentation.cartodb.com/viz/6c1f86e4-ab04-11e4-9c80-0e853d047bba/embed_map' allowfullscreen webkitallowfullscreen mozallowfullscreen oallowfullscreen msallowfullscreen></iframe>
+
+### Finding Distance from Road
+
```
Comment (michellechandra) [inline courses/04-sql-postgis/lesson-2.md]: Include a photo here and description of the map we will be making so people have a visual moving forward.

File: courses/04-sql-postgis/lesson-2.md [hunk 7]
```diff
@@ -2,12 +2,307 @@
...
+  d ASC
+{% endhighlight %}
+
+You could visualize the data in this newly created table by making a choropleth on the column `d`. Since we already did something very similar to that in [Lesson 1](http://academy.cartodb.com/courses/04-sql-postgis/lesson-1.html), we'll move on to new functionality.
+
+![ST_Distance]({{site.baseurl}}/img/course4/lesson2/stdistance.png)
+
+### Visualizing Lines from Musicians to the Road
+
+`ST_MakeLine()` returns a line geometry given two or more points. When working on a collection of points, it returns the path that an object follows if you specify the order the object visited the points. Check out the documentation for more on this.
```
Comment (michellechandra) [inline courses/04-sql-postgis/lesson-2.md]: “it returns the path that an object follows if you specify the order the object visited the points.” 
This sounds weird to me, not clear what you mean here. 
Also link to documentation is missing here.

**Comment by michellechandra** [comment]
@andrewxhill Yeah, I pushed some changes and @ohasselblad and I chatted that I would make comments on the repo regarding changes that were more conceptual/bigger picture.

File: courses/04-sql-postgis/lesson-2.md (hunk 1 above)
Comment (matallo) [inline courses/04-sql-postgis/lesson-2.md]: +1 I had no idea what this was about

File: courses/04-sql-postgis/lesson-2.md (hunk 2 above)
Comment (matallo) [inline courses/04-sql-postgis/lesson-2.md]: +1

File: courses/04-sql-postgis/lesson-2.md (hunk 3 above)
Comment (matallo) [inline courses/04-sql-postgis/lesson-2.md]: +1

File: courses/04-sql-postgis/lesson-2.md (hunk 4 above)
Comment (matallo) [inline courses/04-sql-postgis/lesson-2.md]: worth a link to https://www.youtube.com/watch?v=q65O3qA0-n4 ?

**Comment by matallo** [comment]
looks like `stdwithin.png` is not being used, there's a map instead (better!) shall we remove it?

**Comment by ohasselblad** [comment]
Thanks @michellechandra and @matallo for all the suggestions!
Almost ready for release.. just want @auremoser to take a quick look!

File: courses/04-sql-postgis/lesson-2.md (hunk 6 above)
Comment (ohasselblad) [inline courses/04-sql-postgis/lesson-2.md]: moved map to top of the page

**Comment by auremoser** [comment]
Great lesson! Clever concept :)
* Data link: `http://academy.cartodb.com/d/highway_61_revisited.geojson`
should be `http://academy.cartodb.com/d/highway_61.geojson` ?
* You mention the WHERE clause in the "while it might seem.." part but there is no "where" in the highlighted sql
* No one cares about this, but maybe change "let's take a different tact..." to "let's take a different tack" [because](http://grammarist.com/usage/tack-tack/)
* You probably already explained ESPG 3857 and SRID elsewhere, but if not, maybe linking some info about the latter would be nice when it's mentioned in the highlighted sql:
`boolean ST_DWithin(geometry g1, geometry g2, double precision distance_of_srid)`
* I love that you incorporated Santana's stats at the end! Great that you have the CSS quoted, maybe explain why you binned the colors that way, or was it the default?
Feel free to change anything or not and I'll totes merge in the AM :)

**Comment by ohasselblad** [comment]
Ok @matallo, this is all ready to go!

**Comment by andrewxhill** [comment]
@vuxita what do you think about doing some Google campaigns around 'sql gis' or 'sql codeacademy' or 'learn sql' stuff like that?

**Comment by ohasselblad** [comment]
@matallo we're ready for a merge on this. could you take a look today? thanks!

**Comment by vuxita** [comment]
Hey @andrewxhill - I will have a look and see if we can launch a campaign from March. Will evaluate kw and so on.

**Comment by matallo** [comment]
you are awesome @ohasselblad, thanks!

**Pull request merged**